docs/build-results-archive.md

```diff
@@ -0,0 +1,52 @@
+# Build results archive.
+
+This document describes a build results archive format which can be used to store the sarif results of a build-and-test procedure.
+
+The archive will take the form of a compressed (zstd if available, or bzip2 otherwise) tar file containing the following files.
+
+```
+./build-results-archive
+./build/**/*.sarif
+./colcon-build-cmd
+./colcon-test-cmd
+./processed.sarif
+./vcs-export-exact.repos
+```
+
+The individual files are further described:
+
+## build-results-archive
+
+This is a JSON file containing metadata regarding the current archive.
+The file _must_ contain an single top-level JSON object and that object _must_ contain the `"version"` key with the value of `1`.
+Future revisions of this format may add additional valid versions.
+
+## build/**/*.sarif
+
+The build/ subdirectory _must_ contain the unaltered sarif content of the workspace build directory after the end of the latest colcon test run.
+
+## colcon-build-cmd
+
+This file _must_ contain the full colcon build command described in the colcon logs under the `latest_build` symlink.
+The file, when run as a /bin/sh program or prefixed with the `exec` shell command in a workspace root should invoke colcon in the local environment using the same arguments as the original build.
```

Comment:
Nitpick: `should`, or `shall`?

Reply:
Won't repeat the review comment, but there are more instances of `should` in the doc to check. Also either all {should, shall, must} instances should be wrapped with `_`, or none of them (though happy to be overridden on this one).

Reply:
Good catch on the should vs shall/must, I fixed that in https://github.com/space-ros/process_sarif/pull/6/commits/f3144142872728a02d0d7281cd6ae42dcaf9a5d0 along with emphasizing RFC words consistently.